src/briefcase/integrations/subprocess.py

```diff
@@ -88,6 +89,46 @@ def get_process_id_by_command(
     return None
 
 
+def pause_console_control_if(sub_method):
+    """Decorator for Subprocess methods to conditionally remove dynamic console
+    elements such as the Wait Bar prior to running the subprocess command.
+
+    :param sub_method: wrapped Subprocess method
+    """
+
+    @wraps(sub_method)
+    def inner(sub, cmdline, **kwargs):
+        """Evaluate whether conditions are met to remove any dynamic elements
+        in the console before returning control to Subprocess.
+
+        :param sub: Subprocess object
+        :param cmdline: list of implicit strings that will be run as subprocess command
+        :return: the return value for the Subprocess method
+        """
+        # Just run the command if no dynamic elements are active
+        if not sub.tools.input.is_console_controlled:
+            return sub_method(sub, cmdline, **kwargs)
+
+        remove_dynamic_elements = False
+
+        # Batch (.bat) scripts on Windows.
+        # If cmd.exe is interrupted with CTRL+C while running a bat script,
+        # it may prompt the user to abort the script and dynamic elements
+        # such as the Wait Bar can hide this message from the user.
+        if not remove_dynamic_elements:
```

Comment:
Is this if redundant? I can't see any way that `remove_dynamic_elements` could be `true` here, given that `remove_dynamic_elements` is a local that was set on L112.

Reply:
This check is redundant right now. I added it for forward safety. If multiple checks are done here in the future, we don't want a check to reverse the decision of a previous check to turn off the Wait Bar. This is intended to help ensure that's apparent.

I can remove it if that isn't a particularly compelling rationale.

Reply:
I figured this was probably the intended reason; but given the logic is fairly straightforward, easy to add back in if we need it, and there's no obvious candidate for another "suspend waitbar" use case (that I can think of), I'm inclined to say remove it for now and add it back in if we need it later.